scripts/gen-edpm-baremetal-kustomize.sh

```diff
@@ -123,6 +123,47 @@ EOF
     done
 fi
 
+if [ "${EDPM_SERVER_ROLE}" == "compute" ]; then
+# Create a nova-custom service with a reference to nova-extra-config CM
+cat <<EOF >>kustomization.yaml
+- target:
+    kind: OpenStackDataPlaneService
+    name: nova
+  patch: |-
+    - op: replace
+      path: /metadata/name
+      value: nova-custom
+    - op: add
+      path: /spec/configMaps
+      value:
+        - nova-extra-config
+EOF
+
+# Create the nova-extra-config CM based on the provided config file
+cat <<EOF >>kustomization.yaml
+configMapGenerator:
+- name: nova-extra-config
+  files:
+    - 25-nova-extra.conf=${EDPM_EXTRA_NOVA_CONFIG_FILE}
+  options:
+    disableNameSuffixHash: true
+EOF
+
+# Replace the nova service in the nodeset with the new nova-custom service
+#
+# NOTE(gibi): This is hard to do with kustomize as it only allows
+# list item replacemnet by index and not by value, but we cannot
+# be sure that the index is not changing in the future by
+# adding more services or splitting existing services.
+# The kustozmization would be something like:
+#     - op: replace
+#      path: /spec/services/11
+#      value: nova-custom
+#
+# So we do a replace by value with yq (assuming golang implementation of yq)
+yq -i '(.spec.services[] | select(. == "nova")) |= "nova-custom"' *openstackdataplanenodeset*.yaml
+fi
+
```

Comment:
afar as I see this is the same as https://github.com/openstack-k8s-operators/install_yamls/blob/main/scripts/gen-edpm-kustomize.sh#L197-L236. Could you please put this logic to a common place instead of duplicating it?

Reply:
ack. will update it. Thanks.